agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/JsonConfigPolling.java

```diff
@@ -31,58 +31,56 @@
 import com.microsoft.applicationinsights.TelemetryConfiguration;
 import com.microsoft.applicationinsights.agent.bootstrap.MainEntryPoint;
 import com.microsoft.applicationinsights.agent.bootstrap.configuration.Configuration;
+import com.microsoft.applicationinsights.agent.bootstrap.configuration.Configuration.Sampling;
 import com.microsoft.applicationinsights.agent.bootstrap.configuration.ConfigurationBuilder;
-import com.microsoft.applicationinsights.agent.bootstrap.configuration.InstrumentationSettings.FixedRateSampling;
 import com.microsoft.applicationinsights.internal.util.ThreadPoolUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import static java.util.concurrent.TimeUnit.SECONDS;
 
-public final class JsonConfigWatcher {
+public class JsonConfigPolling implements Runnable {
 
-    private static volatile Long lastModifiedTime = MainEntryPoint.getLastModifiedTime();
-    private static final Logger logger = LoggerFactory.getLogger(JsonConfigWatcher.class);
+    private volatile Long lastModifiedTime = MainEntryPoint.getLastModifiedTime();
```

Comment:
instead of pulling this from MainEntryPoint, i think cleaner to push lastModifiedTime into `pollJsonConfigEveryMinute(long)` method, and then push from there into `JsonConfigPolling` constructor

Reply:
the idea is that makes JsonConfigPolling self-contained, easier to test, and understand on it's own